superset/alfred/utils.py

```diff
@@ -0,0 +1,350 @@
+            ids_string = "','".join(ids)
+        else:
+            raise Exception("No field 'email_ids' found in request")
+        if dates:
+            logger.info(f'Found dates: "{ids_string}" in arguments')
+            for date in dates:
+                try:
+                    datetimes.append(datetime.strptime(date, "%m-%d-%Y"))
+                except ValueError as ve:
+                    logger.info(ve)
+                    continue
+        if alfred_env:
+            logger.info(f'Found Alfred-env: "{alfred_env}" in arguments')
+        else:
+            raise Exception("No field 'alfred_env' found in request")
+
+        logger.info("Generating access token for trino...")
+        trino_app = client.get_oauth_client("trino")
+
+        trino_acces_token, trino_ctx = client.on_behalf_of(
+            f"{trino_app.audience}/{trino_app.scope}",
+            access_token,
+            token_client_name="superset",
+        )
+
+        trino_host = "trino-stg.hogwarts.pb.azure.chimera.cyber.gc.ca"
+        logger.info(f"Establishing connection to trino at {trino_host}...")
+        conn = connect(
+            auth=JWTAuthentication(trino_acces_token),
+            http_scheme="https",
+            host=trino_host,
+            port=443,
+            catalog="hogwarts_pb",
+        )
+
+        sql = f"""
+    SELECT id, urls, "from", to, message_id, reply_to, stream_id, subject, eml_path, bcc, cc, ministerial_authorization, classification, time
+    FROM  hogwarts_pb.harmonized.eml_metadata
+    WHERE id in ('{ids_string}')
+    """
+        if dates:
+            sql += " AND "
+            for i in range(len(dates)):
+                if i > 0:
+                    sql += " OR "
+                sql += f"(time > TIMESTAMP '{datetimes[i].strftime('%Y-%m-%d')}' AND time < TIMESTAMP '{(datetimes[i] + timedelta(days=1)).strftime('%Y-%m-%d')}')"
+        logger.info(f"Querying EML Data from trino at {trino_host}...")
+        cur = conn.cursor()
+        logger.info(f"Created Connection {cur.query_id}")
+        cur.execute(sql)
+        logger.info(f"executed query {cur.query_id}")
+        columns = list(map(lambda d: d[0], cur.description))
+        results = cur.fetchmany(limit + 1)
+        logger.info("Received response...")
+        logger.info(results)
+        processed_rows = []
+
+        logger.info("ROWS")
+        for row in results:
+            logger.info(row)
+            processed_rows.append(sanitize_results(row))
+
+        json_rows = pd.DataFrame.from_records(processed_rows, columns=columns).to_json(
+            orient="records"
+        )
+        logger.info("JSON ROWS")
+        logger.info(json_rows)
+        data_to_map = json.loads(json_rows)
+        logger.info("DATA")
+        logger.info(data_to_map)
+
+        rationale = "Malicious Activity"
+        classification = "PB//CND"
+        cccs_formatting = True  # Extra formatting to fit CCCS standards
```

Comment:
This could potentially be set to False in the other method. Should False be an option there if it's hard-coded to True here? Just checking.

Reply:
removing it as it defaults to true, if there is a case in the future where we want to set it to false it will be easy to change